components/net/filemanager_thread.rs

```diff
@@ -279,21 +219,39 @@ impl FileManagerStore {
         }
     }
 
-    fn select_file<UI>(&self,
-                       patterns: Vec<FilterPattern>,
-                       sender: IpcSender<FileManagerResult<SelectedFile>>,
-                       origin: FileOrigin,
-                       opt_test_path: Option<String>,
-                       ui: &UI)
-        where UI: UIProvider,
-    {
+    #[cfg(any(target_os = "macos", target_os = "linux", target_os = "windows"))]
```

Comment:
I don't think we have to make a distinction between the platforms here. Let send a message to the embedder no matter the platform, and let the embedder decide what to do depending on the platform.